### DISCUSSION TITLE
Batch Jobs: how to know you're in last batch?

### DISCUSSION THREAD

**Discussion by janhenckens** [craftcms/cms#13878] [Q&A] [answered]
I'm working on a plugin that uses the new batched queue jobs that were added in 4.4 and I'm looking for a way to execute a finishing step outside the my batched query when all data has been processed.

Should I try to calculate this myself (by getting a count of the items in my query and the slices in the batch maybe?) or is there a recommended way to go about this?

**Comment by brandonkelly** [comment, marked as answer]
You could do this by overriding the `execute()` function like so:

```php
public function execute($queue): void
{
    parent::execute($queue);

    // Was this the last batch?
    if ($this->itemOffset >= $this->totalItems()) {
        // ...
    }
}
```

**Reply by janhenckens** [reply]
That works, thanks for helping me along!
